101-function-app-storage-private-endpoints/prereqs/prereq.azuredeploy.json

```diff
@@ -0,0 +1,210 @@
+      }
+    },
+    "identityRoleName": {
+      "type": "string",
+      "defaultValue": "[concat('managed-identity-use-deployment-script', uniqueString(resourceGroup().id))]",
+      "metadata": {
+        "description": "Configure least privilege for the managed identity to use the deployment script"
+      }
+    },
+    "principalRoleName": {
+      "type": "string",
+      "defaultValue": "[concat('deployment-principal-run-deployment-script', uniqueString(resourceGroup().id))]",
+      "metadata": {
+        "description": "Configure least privilege for the deployment principal to run the deployment script"
+      }
+    },
+    "identityRoleDescription": {
+      "type": "string",
+      "defaultValue": "Configure least privilege for the managed identity to use the deployment script",
+      "metadata": {
+        "description": "Detailed description of the custom role definition for the managed identity"
+      }
+    },
+    "principalRoleDescription": {
+      "type": "string",
+      "defaultValue": "Configure least privilege for the deployment principal in deployment script",
+      "metadata": {
+        "description": "Detailed description of the custom role definition for the deployment principal"
+      }
+    },
+    "identityActions": {
+      "type": "array",
+      "defaultValue": [
+        "Microsoft.Web/serverfarms/*",
+        "Microsoft.Web/sites/*"
+      ],
+      "metadata": {
+        "description": "Array of actions that the user identity needs to run the deployment script to perform actions upon the Function App and Elastic Premium Plan."
+      }
+    },
+    "principalActions": {
+      "type": "array",
+      "defaultValue": [
+        "Microsoft.Storage/storageAccounts/*",
+        "Microsoft.ContainerInstance/containerGroups/*",
+        "Microsoft.Resources/deployments/*",
+        "Microsoft.Resources/deploymentScripts/*"
+      ],
+      "metadata": {
+        "description": "Array of actions for the custom roleDefinition for the deployment principal to run the deployment script"
+      }
+    },
+    "notActions": {
+      "type": "array",
+      "defaultValue": [],
+      "metadata": {
+        "description": "Array of notActions for the custom roleDefinitions"
+      }
+    },
+    "assignableScope": {
+      "type": "string",
+      "defaultValue": "[resourcegroup().id]",
+      "metadata": {
+        "description": "The scope of the custom role definitions"
+      }
+    }
+  },
+  "variables": {
+    "identityCustomRoleDefName": "[guid(resourcegroup().id, 'identityDeploymentRole')]",
+    "principalCustomRoleDefName": "[guid(resourcegroup().id, 'principalDeploymentRole')]",
+    "identityCustomRoleAssignmentName": "[guid(resourcegroup().id, 'identityDeploymentRoleAssignment')]",
```

Comment:
the roleAssignmentNames need to be idempotent and sufficiently unique... use a combination of principal (objectId or resourceId of the principal), roleDef and scope.